File: 101/exercises/easy5/letterswap.rb

def swap(str)
  words = str.split(' ')
  words.map! do |word|
    word[0], word[-1] = word[-1], word[0]
    word
  end
  words.join(' ')
end

## Tests

p swap('Oh what a wonderful day it is') == 'hO thaw a londerfuw yad ti si'
p swap('Abcde') == 'ebcdA'
p swap('a') == 'a'
p swap('') == ''

File: 210/small-problems/easy3/letter-swap.js
function swap(text) {
  return text.split(' ')
             .map(swapFirstAndLastLetter)
             .join(' ');
}

function swapFirstAndLastLetter(string) {
  if (string.length < 2) return string;

  return string[string.length - 1] + string.slice(1, -1) + string[0];
}

// Tests

console.log(swapFirstAndLastLetter('Oh') === 'hO');
console.log(swapFirstAndLastLetter('what') === 'thaw');
console.log(swapFirstAndLastLetter('a') === 'a');

console.log(swap('Oh what a wonderful day it is') === 'hO thaw a londerfuw yad ti si');
console.log(swap('Abcde') === 'ebcdA');
console.log(swap('a') === 'a');
